File: rmt/src/test/scala/de/flower/rmt/service/ResponseManagerTest.scala
package de.flower.rmt.service

import de.flower.rmt.test.AbstractIntegrationTests
import de.flower.rmt.model.event.Event
import org.testng.Assert._
import org.testng.annotations.Test
import de.flower.rmt.model.{Response, Player, RSVPStatus}
import java.util.List
import scala.collection.JavaConversions._
import org.springframework.beans.factory.annotation.Autowired

/**
 *
 * @author flowerrrr
 */

class ResponseManagerTest extends AbstractIntegrationTests {

    @Test
    def testFindNotResponded() {
        var event: Event = testData.createEvent()
        var players: List[Player] = event.getTeam.getPlayers()
        assertTrue(!players.isEmpty)
        var notResponder = playerManager.findNotResponded(event)
        assertEquals(notResponder, players)

        responseManager.respond(event, players.get(0), RSVPStatus.ACCEPTED, "some comment")
        notResponder = playerManager.findNotResponded(event)
        assertEquals(players.size(), notResponder.size() + 1)
    }

    @Test
    def testFindResponder() {
        var event: Event = testData.createEvent()
        val players = event.getTeam().getPlayers()
        assertTrue(responseManager.findByEventAndStatus(event, RSVPStatus.ACCEPTED).isEmpty())
        var response = responseManager.respond(event, players.get(0), RSVPStatus.ACCEPTED, "some comment")
        assertEquals(response, responseManager.findByEventAndStatus(event, RSVPStatus.ACCEPTED).get(0))

        response = responseManager.respond(event, players.get(1), RSVPStatus.ACCEPTED, "some comment")
        assertTrue(responseManager.findByEventAndStatus(event, RSVPStatus.ACCEPTED).size() == 2)

    }

    @Test
    def testRespond() {
        val event = testData.createEvent()
        val player = event.getTeam().getPlayers().get(0)
        var comment = "Comment #1"
        var status = RSVPStatus.ACCEPTED
        // first initial response
        var response = responseManager.respond(event, player, status, comment)
        assertEquals(response.getStatus(), status)
        assertEquals(response.getComment(), comment)

        // update response for player
        comment = "Comment #2"
        status = RSVPStatus.DECLINED
        response = responseManager.respond(event, player, status, comment)
        assertEquals(response.getStatus(), status)
        assertEquals(response.getComment(), comment)

    }

}

File: rmt/src/test/java/de/flower/rmt/service/ResponseManagerTest.java
package de.flower.rmt.service;

import de.flower.rmt.model.db.entity.Invitation;
import de.flower.rmt.model.db.entity.Player;
import de.flower.rmt.model.db.entity.event.Event;
import de.flower.rmt.model.db.type.RSVPStatus;
import de.flower.rmt.test.AbstractRMTIntegrationTests;
import org.testng.annotations.Test;

import java.util.List;

import static org.testng.Assert.*;

/**
 * @author flowerrrr
 */

public class ResponseManagerTest extends AbstractRMTIntegrationTests {

    @Test
    public void testFindResponder() {
        Event event = testData.createEvent();
        List<Player> players = event.getTeam().getPlayers();
        assertTrue(invitationManager.findAllByEventAndStatus(event, RSVPStatus.ACCEPTED).isEmpty());
        Invitation invitation = responseManager.respond(event.getId(), players.get(0).getUser().getId(), RSVPStatus.ACCEPTED);
        List<Invitation> invitations = invitationManager.findAllByEventAndStatus(event, RSVPStatus.ACCEPTED);
        assertEquals(invitation, invitations.get(0));
        assertEquals((long) invitationManager.numByEventAndStatus(event, RSVPStatus.ACCEPTED), invitations.size());

        invitation = responseManager.respond(event.getId(), players.get(1).getUser().getId(), RSVPStatus.ACCEPTED);
        invitations = invitationManager.findAllByEventAndStatus(event, RSVPStatus.ACCEPTED);
        assertTrue(invitations.size() == 2);
        assertEquals((long) invitationManager.numByEventAndStatus(event, RSVPStatus.ACCEPTED), invitations.size());
    }

    @Test
    public void testRespond() {
        Event event = testData.createEvent();
        Player player = event.getTeam().getPlayers().get(0);
        RSVPStatus status = RSVPStatus.ACCEPTED;
        // first initial invitation
        Invitation invitation = responseManager.respond(event.getId(), player.getUser().getId(), status);
        assertEquals(invitation.getStatus(), status);

        // update invitation for player
        status = RSVPStatus.DECLINED;
        invitation = responseManager.respond(event.getId(), player.getUser().getId(), status);
        assertEquals(invitation.getStatus(), status);
    }
}